components/motor/gpio/basic.go

```diff
@@ -186,39 +183,51 @@ func (m *Motor) setPWM(ctx context.Context, powerPct float64, extra map[string]i
 		errs = multierr.Combine(errs, m.EnablePinHigh.Set(ctx, true, extra))
 	}
 
+	m.powerPct = powerPct
+	powerPct = math.Min(powerPct, m.maxPowerPct)
+	powerPct = math.Max(powerPct, -1*m.maxPowerPct)
```

Comment:
the powerPCt logic in line 205 needs power percent to be negative to flip the AB pins in the SetPower call, and set the PMW on one pin to pwmPct, and the other to the remainder. I get a failing test if i do this because the calculation is wrong and sets the pct to 0 on the wrong pin.

Reply:
ah okay makes sense. do we not care about checking the minimum then?

Reply:
no since this function has always had a functional minimum of 0.01 now that I'm reading it.

Reply:
but added that check back, what's going on now is that even if you have a minimum power, the default is to turn off the motor if it's less than 0.001 power in that switch statement. So we have a hard coded minimum.

Reply:
but I assume if the minimum is like 20% all values below that should be forced to 20% since that's what we're doing with the maximum. not just shut off if it's below the minimum?